pinot-core/src/main/java/org/apache/pinot/core/operator/filter/AndFilterOperator.java

```diff
@@ -45,6 +47,39 @@ protected FilterBlock getNextBlock() {
     return new FilterBlock(new AndDocIdSet(filterBlockDocIdSets));
   }
 
+  @Override
+  public boolean canOptimizeCount() {
+    boolean allChildrenCanProduceBitmaps = true;
+    for (BaseFilterOperator child : _filterOperators) {
+      allChildrenCanProduceBitmaps &= (child.isResultMatchingAll() | child.canProduceBitmaps());
```

Comment:
1. I don't see merit in terminating early
2. I plan to change `BaseFilterOperator.isResultMatchingAll()` to return true when this can be determined by metadata. As far as I'm aware, tautologous predicates (w.r.t. column metadata) are not eliminated, and I have test cases which demonstrate this. Are they eliminated somewhere else? Either way, I can remove this logic for now.

Reply:
I've removed the code for now.

Reply:
I haven't terminated early for a couple of reasons
1. this list having 10 elements would be a monster query
2. it looks ugly
3. it's better for people who use debuggers if there's only one exit point